File: java-selenium/src/main/java/com/framework/pages/InventoryPage.java
package com.framework.pages;

import io.qameta.allure.Step;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

import java.util.List;

public class InventoryPage extends BasePage {

    @FindBy(className = "inventory_item_name")
    private List<WebElement> itemNames;

    @FindBy(className = "inventory_item")
    private List<WebElement> items;

    @FindBy(css = ".shopping_cart_link")
    private WebElement cartLink;

    @FindBy(css = "[data-test='product-sort-container']")
    private WebElement sortDropdown;

    @Step("Verify inventory page is loaded")
    public boolean isLoaded() {
        waitForUrl("inventory");
        return !itemNames.isEmpty();
    }

    @Step("Get inventory item count")
    public int getItemCount() {
        return items.size();
    }

    @Step("Add first item to cart")
    public InventoryPage addFirstItemToCart() {
        WebElement addButton = items.get(0).findElement(
            org.openqa.selenium.By.cssSelector("button[id^='add-to-cart']")
        );
        click(addButton);
        return this;
    }

    @Step("Navigate to cart")
    public void goToCart() {
        click(cartLink);
    }

    @Step("Get first item name")
    public String getFirstItemName() {
        return getText(itemNames.get(0));
    }
}

File: typescript-playwright/src/pages/InventoryPage.ts
import { Page, Locator } from '@playwright/test';
import { BasePage } from './BasePage';

export class InventoryPage extends BasePage {
  readonly inventoryItems: Locator;
  readonly itemNames: Locator;
  readonly cartLink: Locator;
  readonly sortDropdown: Locator;

  constructor(page: Page) {
    super(page);
    this.inventoryItems = page.locator('.inventory_item');
    this.itemNames      = page.locator('.inventory_item_name');
    this.cartLink       = page.locator('.shopping_cart_link');
    this.sortDropdown   = page.locator('[data-test="product-sort-container"]');
  }

  async isLoaded(): Promise<boolean> {
    await this.waitForUrl(/inventory/);
    return this.inventoryItems.count().then((c) => c > 0);
  }

  async getItemCount(): Promise<number> {
    return this.inventoryItems.count();
  }

  async getFirstItemName(): Promise<string> {
    return this.getText(this.itemNames.first());
  }

  async addItemToCartByName(name: string): Promise<void> {
    const item = this.page.locator(`.inventory_item:has-text("${name}")`);
    const addButton = item.locator('button[id^="add-to-cart"]');
    await this.click(addButton);
  }

  async addFirstItemToCart(): Promise<string> {
    const firstName = await this.getFirstItemName();
    const addButton = this.inventoryItems.first().locator('button[id^="add-to-cart"]');
    await this.click(addButton);
    return firstName;
  }

  async goToCart(): Promise<void> {
    await this.click(this.cartLink);
  }

  async sortBy(value: 'az' | 'za' | 'lohi' | 'hilo'): Promise<void> {
    await this.sortDropdown.selectOption(value);
  }
}
